### PR TITLE
feat: add apps showcase section

### PR DESCRIPTION
**Pull request by erickzhao**
Ports over the old https://electronjs.org/apps showcase page to Docusaurus.

Note that this PR drops the dependency to `electron-apps` in favour of pointing directly to a JSON file hosted on GitHub. This file is currently available on my `electron/apps` fork here: https://github.com/erickzhao/apps

I figured using a fork would be easier to work with since I was making many cleanup changes to the showcase data.

Notable differences from the old version:
* Apps no longer have their own page on the website
* No more search bar (could add back if useful)
* New "Favourites" section highlights certain apps (maintainer apps + some others—this is subjective and list could change.)

See preview: https://62fd49260bd1270098b8a106--shimmering-stroopwafel-37e3bb.netlify.app/apps/

image

### CODE REVIEW

**Comment by BlackHole1** [comment]
I would find it very cool if there were some animation effects when the mouse is over the Card (hover event)

**Comment by Kilian** [comment]
The tinted backgrounds are such a nice touch! <3

Showcase doesn't fit on mobile:
62fd49260bd1270098b8a106--shimmering-stroopwafel-37e3bb netlify app_apps Mobile Phone 2022-08-18 07 42 41

Additionally, those number badges are very small (in fact, they're so small they snap to 6px) Instead of font-size 20%, try font-size: 80%:
62fd49260bd1270098b8a106--shimmering-stroopwafel-37e3bb netlify app_apps Small Tablet 2022-08-18 07 52 20


Listing causes an overflow, you probably need to allow it to go down to 1 column:
62fd49260bd1270098b8a106--shimmering-stroopwafel-37e3bb netlify app_apps Overview 2022-08-18 07 43 54

The heart is not centered vertically, you solve that with this code:

current: 62fd49260bd1270098b8a106--shimmering-stroopwafel-37e3bb netlify app_apps Small Tablet 2022-08-18 07 50 15 fixed: 62fd49260bd1270098b8a106--shimmering-stroopwafel-37e3bb netlify app_apps Small Tablet 2022-08-18 07 49 37


```css
.appCard_qOQV.appCardFav_qVYO::before {
  place-items: center;
  backdrop-filter: blur(5px);
  background: rgba(255, 255, 255, 0.7);
  border: 1px solid rgb(245, 245, 245);
  border-radius: 100%;
  box-shadow: var(--shadow-elevation-high);
  content: "❤️";
  display: grid;
  height: 1.75rem;
  left: -0.5rem;
  position: absolute;
  top: -0.5rem;
  width: 2rem;
  padding-top: 0.25rem;
  line-height: inherit;
}
```

It would be nice if the filter stayed in view with `position: sticky`:
Selection_1248
Quick mockup CSS that could use polish:
```css
position: sticky; 
top: 86px; 
background: rgb(255, 255, 255); 
z-index: 2; 
padding-top: 1rem; 
width: calc(100% + 2rem); 
margin-left: -1rem; 
padding-inline: 1rem;
```

I think some sort of "search" (filter down to matching strings and tags?) would be nice, but it's not super needed.

**Comment by Toinane** [comment]
I'm agree with the number badges are really too small, even on my 27" screen. Don't know why, but the numbers are even more small on Firefox than Chrome
image

**Comment by bnb** [comment]
Overall I really like this! Haven't dug into the specifics, but it's really good IMO.

Might want to make VS Code a Favorite :P

**Comment by erickzhao** [comment]
Thank you for the round of feedback everyone!

- [x] Showcase `font-size` decreased on mobile viewports
- [x] Number badge `font-size` increased
- [x] Category `font-weight` constant
- [x] Mobile overflow fixed
- [x] Favourites heart centred vertically
- [x] Image loading strategy changed to `lazy`
- [x] Added VSCode as a favourite app
- [x] Sticky category selector added 

Aside from that, a few more tweaks were made:
* Mobile viewports now use a Dropdown to filter the apps list rather than display all categories to save space.
* Shadows are bit softer site-wide (`--shadow-color` variable changed from `0deg 0% 0%` to `0deg 0% 63%`).

> I would find it very cool if there were some animation effects when the mouse is over the Card (hover event)

@BlackHole1 I have concerns about adding hover animations because they usually indicate that the element is clickable. Do you have an example that you think would work well here?

**Comment by BlackHole1** [comment]
> I have concerns about adding hover animations because they usually indicate that the element is clickable

@erickzhao Hmm. Sorry, I hadn't thought of that before. I checked out sites like hugo and found that this is indeed the case. 🙇‍♂️
So let's keep it the same.

File: src/pages/apps/apps.module.scss
Comment (BlackHole1) [inline src/pages/apps/apps.module.scss:50]:
```suggestion
.filterBadge {
  margin-left: 0.2rem;
  padding: 0.2rem 0.4rem;
  font-size: 80%;
  background-color: var(--ifm-color-emphasis-200);
  border-color: var(--ifm-color-emphasis-200);
  color: var(--ifm-color-content);
}
```

Optimising the colour scheme of Badge in dark mode

Before:

https://user-images.githubusercontent.com/8198408/186053930-3da9db47-a1a2-4c9e-b02d-4e017e853544.mp4

After:

https://user-images.githubusercontent.com/8198408/186054146-3074a568-1ba8-4046-acf9-9c1fb745a9b4.mp4

File: src/pages/apps/index.jsx
Comment (BlackHole1) [inline src/pages/apps/index.jsx:119]:
```suggestion
              <button
                className={clsx(
                  'button button--secondary',
                  styles.sortDropdownButton
                )}
              >
```

```scss
// File: src/pages/apps/apps.module.scss
.sortDropdownButton {
  color: var(--ifm-color-content) !important;
  background-color: var(--ifm-color-emphasis-200);
  border-color: var(--ifm-color-emphasis-200);
}
```

Before:

https://user-images.githubusercontent.com/8198408/186056443-214da4d7-bef1-43bc-9f18-7ef98cbddd6a.mp4

After:

https://user-images.githubusercontent.com/8198408/186056463-3b5a4b4e-be42-4a51-a416-a75b4fc0b120.mp4

File: src/pages/apps/index.jsx
Comment (BlackHole1) [inline src/pages/apps/index.jsx:122]:
```suggestion
              <ul className={clsx('dropdown__menu', styles.sortDropdownMenu)}>
```

```scss
// File: src/pages/apps/apps.module.scss
.sortDropdownMenu {
  min-width: 100%;
}
```

Before:
image


After:
image

File: src/plugins/apps/index.js
Comment (BlackHole1) [inline src/plugins/apps/index.js:18]:
```suggestion
        'agora-flat',
```
The full name of the product is `Agora Flat`

Comment (BlackHole1) [reply src/plugins/apps/index.js:18]: Sorry. This is an issue I accidentally introduced earlier and I have submitted a fix PR to apps. 🙇‍♂️

Ref: https://github.com/electron/apps/pull/2049

File: src/pages/apps/apps.module.scss
Comment (erickzhao) [reply src/pages/apps/apps.module.scss:28]: Good catch. I don't think I can use `--ifm-background-color` directly if I want transparency, though.

File: src/pages/apps/apps.module.scss
Comment (BlackHole1) [inline src/pages/apps/apps.module.scss:143]:
```suggestion
  border-color: var(--ifm-color-emphasis-200);
  
  &:hover {
    color: var(--ifm-link-color);
  }
```

Before:
image
image


After:
image
Image2

**Review by BlackHole1 — APPROVED** [review]
LGTM

**Review by VerteDinde — APPROVED** [review]
Looks awesome!

File: .gitignore
Comment (VerteDinde) [inline .gitignore:12]: Nit, but do you want to include `erick` in the ignore? Wasn't sure if that was just for testing

Comment (erickzhao) [reply .gitignore:12]: lol

File: src/plugins/apps/index.js
Comment (BlackHole1) [inline src/plugins/apps/index.js:7]: Do we need to wait for you to submit a PR to electron/apps and merge it before merging this PR?

**Review by Toinane — APPROVED** [review]
Looks really good! :+1:

**Pull request merged**
